### ISSUE TITLE
Dimensionless vs. without unit

### ISSUE THREAD

**Issue by ThibHlln** [NCAS-CMS/cf-python#205]
Hi, 

I am wondering if the first case below should not return `True` instead?
```python
>>> cf.Data(1, '1').has_units()  # dimensionless
False
>>> cf.Data(1).has_units()  # missing unit
False
```

since `cf.Data(1, '1').get_units()` returns `'1'` while `cf.Data(1).get_units()` raises a `ValueError` .

I think it could be useful if it were to return `True` as this would allow to distinguish between data explicitly defined dimensionless, as opposed to data without unit.

```python
>>> cf.environment(paths=False)
Platform: macOS-10.16-x86_64-i386-64bit
HDF5 library: 1.10.6
netcdf library: 4.7.4
udunits2 library: /bin/../lib/libudunits2.dylib
Python: 3.8.5
netCDF4: 1.5.5.1
cftime: 1.3.1
numpy: 1.19.5
psutil: 5.8.0
scipy: 1.6.0
matplotlib: 3.3.3
ESMF: 8.0.1
cfdm: 1.8.8.0
cfunits: 3.3.1
cfplot: 3.0.38
cf: 3.8.0
```

**Comment by davidhassell** [comment]
Hi Thibault,

This is a bug resulting from not modifying the behaviour from the imported cfdm library. This has been fixed for 3.9.0 (details to follow), which will be released "soon, in May" - we're just reviewing the final PRs prior to the public release.

Thanks for spotting it!
David

**Comment by davidhassell** [comment]
I don't have a separate PR for this, but the changes are in  https://github.com/davidhassell/cf-python/commit/c11bce0395e820595ca28de23e258b495407a667

**Comment by ThibHlln** [comment]
Thank you David!

While waiting for v3.9.0, I am using the following to do the job:
```python
>>> cf.Data(1, '1').get_units(False)
'1'
>>> cf.Data(1).get_units(False)
False
```

**Comment by davidhassell** [comment]
Fixed with the release of 3.9.0
